packages/inertia/src/inertia.js

```diff
@@ -278,30 +337,32 @@ export default {
   },
 
   remember(data, key = 'default') {
-    let newState = { ...window.history.state }
-    newState.cache = newState.cache || {}
-    newState.cache[key] = data
-    this.replaceState(newState)
+    let page = { ...this.page }
+    page.rememberedState[key] = data
+    this.replaceState(page)
```

Comment:
Actually here if `rememberedState` is an object, you're mutating the original `this.page.rememberedState`, because your clone action is only on the first level.

You may need to do something like this:
```suggestion
    this.replaceState({
      ...this.page,
      rememberedState: {
    	...this.page.rememberedState,
    	[key]: data,
   	  }
    })
```

Reply:
Good catch! 👍